src/util/configurations.rs

```diff
@@ -0,0 +1,83 @@
+//! This is the config struct used to pass in configurations into the channel. This is kept by the channel manager
+//! channel_limits is only passed in as a reference and only used when accepting or opening a channel
+//! channel options is cloned into channel because those are all specified unique channel settings. It uses channel managers one as a base
+
+/// This is the main user config
+/// This struct should contain all user customizable options as this is passed to the channel to be accessed
+#[derive(Clone, Debug)]
+pub struct UserConfig{
+	/// optional user specified channel limits
+	/// These are only used on startup of channels, and are referenced to a single instance
+	pub channel_limits : ChannelHandshakeLimits,
+	/// channel options are unique to each channel
```

Comment:
I think in the future we may want to pass channel_limits into the new-outbound-channel function as well, so everything is unique to channels, at least optionally, the difference is this one can be changed (or at least represents things that matter) at runtime.